erpnext/controllers/accounts_controller.py

```diff
@@ -929,6 +930,25 @@ def get_gl_dict(self, args, account_currency=None, item=None):
 
 		return gl_dict
 
+	def get_voucher_subtype(self):
+		voucher_subtypes = {
+			"Journal Entry": "voucher_type",
+			"Payment Entry": "payment_type",
+			"Stock Entry": "stock_entry_type",
+			"Asset Capitalization": "entry_type",
+		}
+		if self.doctype in voucher_subtypes:
+			return self.get(voucher_subtypes[self.doctype])
+		elif self.doctype == "Purchase Receipt" and self.is_return:
+			return "Purchase Return"
+		elif self.doctype == "Delivery Note" and self.is_return:
+			return "Sales Return"
+		elif (self.doctype == "Sales Invoice" and self.is_return) or self.doctype == "Purchase Invoice":
+			return "Credit Note"
+		elif (self.doctype == "Purchase Invoice" and self.is_return) or self.doctype == "Sales Invoice":
+			return "Debit Note"
+		return self.doctype
```

Comment:
Do we need to save this? GL entry table is already huge it will even more redundant information after this :grimacing:

Reply:
Initially, thought of directly determining the Voucher Subtype while displaying the General Ledger Report. However, that would require running the conditional for each row. Again this would add to the amount of processing needed there. What would be a better approach?

Reply:
A `get_value` for fetching different doc fields for different vouchers would also be required for every row right along with the `is_return` condition?